File: src/java/EntityFX/src/EntityFX/Core/Generic/HashBenchmark.java
package EntityFX.Core.Generic;


import java.io.FileNotFoundException;

import EntityFX.Core.Writer;

public class HashBenchmark extends HashBase<byte[]> {
    public HashBenchmark(Writer writer, boolean printToConsole) throws FileNotFoundException {
        super(writer, printToConsole);
    }

    @Override
    public byte[] benchImplementation() {
        byte[] result = new byte[] { };
        for (int i = 0; i < Iterrations; i++) {
            result = doHash(i, arrayOfBytes);
        }
        return result;
    }
}

File: src/python/entityfx/hash_benchmark.py

from entityfx.hash_base import HashBase

class HashBenchmark(HashBase):
    
    def benchImplementation(self) -> bytearray:
        result = None

        for i in range(0, self._iterrations):
            result = HashBase._doHash(i, self._array_of_bytes )
        return result